### PR TITLE
Fix XLA tensor views which are not handled by the XLA tensor core.

### LINKED ISSUE #514: assignment issue
Minimal repro:
```
import torch
import torch_xla
import torch_xla_py.utils as xu
import torch_xla_py.xla_model as xm
import pdb
xla_device = xm.xla_device()
d = torch.ones((10, 4)).to(xla_device)
d[:, 0::2] = 2
pdb.set_trace()
print(d)
```
expected:
```
tensor([[2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.],
        [2., 1., 2., 1.]])
```
but got all ones.

### PR DESCRIPTION
**Pull request by dlibenzi**
Fixes #514

### CODE REVIEW

File: torch_xla/csrc/tensor.cpp [hunk 1]
```diff
@@ -531,21 +517,26 @@ void XLATensor::SetScalarType(
...
 
-void XLATensor::DiscardXlaData() {
-  XLA_CHECK(data()->tensor_data);
+void XLATensor::MakeWriteableTensorDataSource() {
+  c10::optional<at::Tensor> tensor_data = CurrentTensorData();
+  XLA_CHECK(tensor_data);
+  if (data()->view != nullptr) {
+    // We are handing out an at::Tensor which the callers might write into it.
+    // If this is a view, we need to push a TensorData update to the view. A
+    // TensorData node will fetch the status of the at::Tensor at lowring time,
```
Comment (asuhan) [inline torch_xla/csrc/tensor.cpp]: lowering

File: torch_xla/csrc/ops/tensor_data.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+#include "torch_xla/csrc/ops/xla_ops.h"
+#include "torch_xla/csrc/tensor_util.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+namespace {
+
+template <typename T>
+void EmitValues(std::ostream& stream, const at::Tensor& tensor) {
```
Comment (asuhan) [inline torch_xla/csrc/ops/tensor_data.cpp]: I'd replace `Emit` with `Print` for both helpers. `Emit` is an ambiguous term.

File: torch_xla/csrc/tensor.cpp (hunk 1 above)
Comment (dlibenzi) [reply torch_xla/csrc/tensor.cpp]: Done.

File: torch_xla/csrc/ops/tensor_data.cpp (hunk 2 above)
Comment (dlibenzi) [reply torch_xla/csrc/ops/tensor_data.cpp]: SG!

**Pull request merged**